src/backend/internal/controller/user/get_user_info.go

```diff
@@ -0,0 +1,42 @@
+package user
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/Roongkun/software-eng-ii/internal/model"
+	"github.com/gin-gonic/gin"
+)
+
+func (r *Resolver) GetUserInfo(c *gin.Context) {
+	UserIdInput := model.UserIdInput{}
```

Comment:
Creating this as a model structure is redundant, as it contains only one field.

In my opinion, these two methods are the simplest ones to resolve this problem:

- You can create a `map[string]interface{}` for the binding. 
```go
var idInput map[string]interface{}
if err := c.BindJSON(&idInput); err != nil {
    c.JSON(400, gin.H{"error": err.Error()})
    return
}

uuid, ok := idInput["user_id"].(string)
if !ok {
    c.JSON(400, gin.H{"error": "UUID not provided or invalid"})
    return
}
```

- You can simply attach the `id` with the endpoint. The endpoint for this could be changed to `/v1/get-user/:id`. You can then use `c.Param("id")` to retrieve the id.